osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerReadyButton.cs

```diff
@@ -155,10 +155,14 @@ private void updateButtonText()
 
                     case MultiplayerUserState.Spectating:
                     case MultiplayerUserState.Ready:
-                        Text = room.Host?.Equals(localUser) == true
+                        Text = multiplayerClient.IsHost
                             ? $"Start match {countText}"
                             : $"Waiting for host... {countText}";
+                        break;
 
+                    // Show the abort button for the host as long as gameplay is in progress.
+                    case MultiplayerUserState when multiplayerClient.IsHost && room.State != MultiplayerRoomState.Open:
```

Comment:
Probably shouldn't be using `case..when` (for @peppy's sake).

Reply:
I've refactored this. There's no world in which I would be able to get behind having a `case` which *doesn't even switch on the fucking switch variable* sorry.